src/dialogs/create-file.tsx

```diff
@@ -0,0 +1,233 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+
+import { Alert, AlertVariant } from '@patternfly/react-core/dist/esm/components/Alert';
+import { Button } from '@patternfly/react-core/dist/esm/components/Button';
+import { Form, FormGroup } from "@patternfly/react-core/dist/esm/components/Form";
+import { FormSelect, FormSelectOption } from "@patternfly/react-core/dist/esm/components/FormSelect";
+import { Modal, ModalVariant } from '@patternfly/react-core/dist/esm/components/Modal';
+import { TextArea } from '@patternfly/react-core/dist/esm/components/TextArea';
+import { TextInput } from "@patternfly/react-core/dist/esm/components/TextInput";
+import { Stack } from '@patternfly/react-core/dist/esm/layouts/Stack';
+
+import cockpit from 'cockpit';
+import { FormHelper } from 'cockpit-components-form-helper';
+import type { Dialogs, DialogResult } from 'dialogs';
+import { useInit } from 'hooks';
+import { superuser } from 'superuser';
+
+import "./editor.scss";
+import { useFilesContext } from '../app.tsx';
+import { checkFilename } from '../common.ts';
+import { get_owner_candidates } from '../ownership.tsx';
+
+import { edit_file } from './editor.tsx';
+
+const _ = cockpit.gettext;
+
+function read_tag(filename: string): Promise<string> {
+    return new Promise(resolve => {
+        cockpit.file(filename, { superuser: "try" }).read()
+                .then((((_content: string, tag: string) => {
+                    resolve(tag);
+                }) as any /* eslint-disable-line @typescript-eslint/no-explicit-any */));
```

Comment:
Uhmm don't follow this? A promise can't return multiple args and we want the tag? Of course I can add a catch(exc) => reject(exc) but you don't want that either?

The :hear_no_evil:  is the same as:
```
    load_file() {
        // Can't do this async because we can't get the tag via await
        this.file.read()
                .then(((content: string, tag: string) => {
                    this.update({ content, tag_now: tag, tag_at_load: tag, error: null });
                }) as any /* eslint-disable-line @typescript-eslint/no-explicit-any */)
                .catch(error => this.update({ error }));
    }
```